addons/sale/models/sale_order_line.py

```diff
@@ -309,6 +309,14 @@ def _compute_name(self):
             name = line._get_sale_order_line_multiline_description_sale()
             if line.is_downpayment and not line.display_type:
                 context = {'lang': line.order_partner_id.lang}
+                invoice = line._get_invoice_lines().move_id
+                if len(invoice) == 1 and invoice.payment_reference and invoice.invoice_date:
+                    name = _(
+                        "%(line_description)s (ref: %(reference)s on %(date)s)",
+                        line_description=name,
+                        reference=invoice.payment_reference,
+                        date=format_date(line.env, invoice.invoice_date),
+                    )
```

Comment:
Mhm, this is potentially confusing since you only want that description when the invoice is posted and you could potentially (or at least conceptually) enter one of the if below.

IMHO, you could either extend `_get_downpayment_state` to get the `posted` status:
```python
    elif dp_state == 'posted':
        invoice = line.invoice_lines
        if len(invoice) == 1 and ...:
            ...
```
or simply add your code in an `else` after the `elif` condition.
That way, the computation of the name would be clearer, with a clear path depending on the invoice state.